netbox/netbox/tables/columns.py

```diff
@@ -235,7 +235,12 @@ def render(self, record, table, **kwargs):
 
         model = table.Meta.model
         request = getattr(table, 'context', {}).get('request')
-        url_appendix = f'?return_url={quote(request.get_full_path())}' if request else ''
+        query_params = request.GET.copy() if request else {}
+        if return_url := query_params.pop('return_url', None):
```

Comment:
I could be missing something but this would this not be sufficient?

```suggestion
        if return_url := request.GET.get('return_url'):
```

Reply:
We can probably simplify this to something like:

```python
return_url = request.GET.get('return_url', request.get_full_path())
url_appendix = f'?return_url={quote(return_url)}' if return_url else ''
```

Reply:
> I could be missing something but this would this not be sufficient?

I tested this using services on device and that worked just it was before. Then did the same test using image attachment and post edit and delete, I landed up on the device page.

While implementing this, I was thinking the same solution should be implemented application wide though.

If you can think of a use-case, that I ay have missed let know.